extensions/api/auth-spi/src/main/java/org/eclipse/dataspaceconnector/api/auth/AuthenticationRequestFilter.java

```diff
@@ -0,0 +1,41 @@
+/*
+ *  Copyright (c) 2020 - 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.api.auth;
+
+import jakarta.ws.rs.container.ContainerRequestContext;
+import jakarta.ws.rs.container.ContainerRequestFilter;
+import org.eclipse.dataspaceconnector.api.exception.NotAuthorizedException;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class AuthenticationRequestFilter implements ContainerRequestFilter {
+    private final AuthenticationService authenticationService;
+
+    public AuthenticationRequestFilter(AuthenticationService authenticationService) {
+        this.authenticationService = authenticationService;
+    }
+
+    @Override
+    public void filter(ContainerRequestContext requestContext) {
+        var headers = requestContext.getHeaders();
+
+        var isAuthenticated = authenticationService.isAuthenticated(headers.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
+
+        if (!isAuthenticated) {
+            throw new NotAuthorizedException();
```

Comment:
shouldn't be `AutenticationFailedException`?

Reply:
I think there is a difference between not being able to perform an authentication (`AuthenticationFailedException`) and the corresponding entity not being _authorized_ (`NotAuthorizedException`).

Reply:
to quote [Wikipedia](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_client_errors):
> Http 401: Similar to 403 Forbidden, but specifically for use when authentication is required and has failed or has not yet been provided

So the 401 is returned when the auth service throws an `AuthenticationFailedException`, but the 403 is returned, when the authentication header was understood but the "server refuses action".